flowmachine/flowmachine/core/server/query_schemas/flows.py

```diff
@@ -49,13 +50,14 @@ def _flowmachine_query_obj(self):
         """
         loc1 = self.from_location._flowmachine_query_obj
         loc2 = self.to_location._flowmachine_query_obj
-        return RedactedFlows(flows=Flows(loc1, loc2))
+        return RedactedFlows(flows=Flows(loc1, loc2, join_type=self.join_type))
 
 
 class FlowsSchema(BaseSchema):
     # query_kind parameter is required here for claims validation
     query_kind = fields.String(validate=OneOf(["flows"]))
     from_location = fields.Nested(InputToFlowsSchema, required=True)
     to_location = fields.Nested(InputToFlowsSchema, required=True)
+    join_type = fields.String(validate=OneOf(Flows.valid_joins))
```

Comment:
Given that `Flows.valid_joins` isn't used within the Flows object, I think I'd prefer the list of valid joins to be directly specified here.

Reply:
I'd also like to have either `required=True` (usually my preference) or `missing="inner"` (possibly preferable in this situation for backward compatibility) here.